src/dashboard/Data/Browser/LoginDialog.react.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import ParseApp from 'lib/ParseApp';
+import PropTypes from 'prop-types';
+import Modal from 'components/Modal/Modal.react';
+import LoginRow from 'components/LoginRow/LoginRow.react';
+import Notification from 'dashboard/Data/Browser/Notification.react';
+
+export default class LoginDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      open: false,
+      username: '',
+      password: ''
+    };
+
+    this.handleOpen = this.handleOpen.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+    this.login = this.login.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+  }
+
+  handleOpen() {
+    this.setState({ open: true });
+  }
+
+  handleClose() {
+    this.setState({ open: false });
+  }
+
+  async login() {
+    const { username, password } = this.state;
+    if (!!username && !!password) {
+      try {
+        await this.props.login(username, password);
```

Comment:
I don't know if this is a side effect but If I stop browsing as a User and try to browse as a user again but leaving the username and password blank. I browser as the previous user I logged out as. Let me know if this make sense.

Reply:
True, it was bug. Fixed it with setting username and password state to '' on dialog close.